libcxx/src/filesystem/path.cpp

```diff
@@ -368,7 +368,8 @@ size_t hash_value(const path& __p) noexcept {
   size_t hash_value = 0;
   hash<string_view_t> hasher;
   while (PP) {
-    hash_value = __hash_combine(hash_value, hasher(*PP));
+    string_view_t Part = PP.inRootDir() ? PATHSTR("/") : *PP;
```

Comment:
So IIUC you canonicalize `<DRIVE>:/` to just `/` for the purpose of `hash_value`?

Reply:
No, the `<DRIVE>:` is [root-name](https://timsong-cpp.github.io/cppwp/n4861/fs.class.path#nt:root-name). Its different from root-directory (which is '/' or '\').
What I did was use '/' to replace the root directory.